engine/src/main/java/org/terasology/rendering/shader/ShaderParametersBase.java

```diff
@@ -43,27 +45,29 @@ public void applyParameters(Material program) {
         program.setFloat("viewingDistance", CoreRegistry.get(Config.class).getRendering().getViewDistance().getChunkDistance().x * 8.0f);
 
         WorldRenderer worldRenderer = CoreRegistry.get(WorldRenderer.class);
-        WorldProvider worldProvider = CoreRegistry.get(WorldProvider.class);
+        BackdropProvider backdropProvider = CoreRegistry.get(BackdropProvider.class);
 
-        if (worldRenderer != null) {
-            program.setFloat("daylight", worldRenderer.getDaylight(), true);
+        if (worldRenderer != null && backdropProvider != null) {
```

Comment:
Good point about 3d elements in menu and loading screen, i keep forgetting
about that eventuality.

It seems to me that the world renderer as it is inadequate for those
purposes. It would make sense to have specific, simplier renderers for
things such as character or even world previews. -Unless- we want a
completely "ingame" look , i.e. to test how my cool camouflaged skin works
in a desert biome or to really see the world looks like before stepping in
it. The decisions in these regards i guess are as much about UX/design as
they technical.

I guess it would be good to reuse as much as possible what we have, i.e.
loading a very small world around a character preview, perhaps using a game
state that doesn't fire up as many systems as the actual game.

But I'm getting off the subject. For the purpose of this PR sounds like we
can leave things as they are. It'd be good to open a couple of issues
though: one regarding a better system for material's parameters and one
about creating a character preview using the current world renderer to
display a largely empty or simplified world.